components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -224,6 +224,10 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             }
 
             String callbackUrl = getCallbackUrl(authenticatorProperties);
+            if (Boolean.parseBoolean((String) context.getProperty(FacebookAuthenticatorConstants.IS_API_BASED))) {
```

Comment:
Can we guarantee that properties getting from the context are not null? If not there can be NPE.

Reply:
Addressed with https://github.com/wso2-extensions/identity-outbound-auth-facebook/pull/61